### PR TITLE
Fixes #19651 - Listing all auth sources

### PR DESCRIPTION
**Pull request by dhlavac**
For testing, you need to add new auth sources API from PR https://github.com/theforeman/foreman/pull/4865

### CODE REVIEW

File: lib/hammer_cli_foreman/auth_source.rb [hunk 1]
```diff
@@ -3,8 +3,22 @@
 module HammerCLIForeman
 
   class AuthSource < HammerCLIForeman::Command
-
+    resource :auth_sources
     subcommand 'ldap', HammerCLIForeman::AuthSourceLdap.desc, HammerCLIForeman::AuthSourceLdap
+
+    desc _('List all auth sources.')
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/auth_source.rb]: This description should be moved into the `ListCommand` class

**Comment by tstrachota** [comment]
[test]

**Comment by dhlavac** [comment]
@tstrachota changed

**Comment by tstrachota** [comment]
The command looks fine. The server part needs to be merged prior to this PR though.

@dhlavac would you mind moving the unit tests to functional ones? We started to prefer testing the commands in the context of the whole cli some time ago.

**Comment by dhlavac** [comment]
@tstrachota Thanks for review, is this necessary to move tests to functional? This commit add only listing option on all auth_sources

**Comment by tstrachota** [comment]
There's a lot more value in the functional tests. They cover the command to a larger extent and provide a lot more readable output. So yes, please move them.

I understand that this commit adds only the listing option. It's fine to write the functional tests only for that list command.

There are some helpers that can make your life easier when you're be testing the command output. See an example here:
https://github.com/theforeman/hammer-cli-foreman/blob/c7fcc147540282b6768b0b277069862c6f51c99c/test/functional/commands/list_test.rb#L102-L139

**Comment by dhlavac** [comment]
@tstrachota  test mooved

**Comment by tstrachota** [comment]
@dhlavac thanks for the updates but your test is executing completely different command. It tests a generic `ListCommand` base class. You should test your specific `AuthSource::ListCommand`

**Comment by tstrachota** [comment]
Please check other tests on `test/functional`. They're not very complicated.

**Comment by tstrachota** [comment]
@dhlavac see https://github.com/tstrachota/hammer-cli-foreman/blob/pr-ofedoren-feat-21514-access-token/test/functional/personal_access_token_test.rb#L34-L53 for inspiration.

**Comment by dhlavac** [comment]
@tstrachota Thanks for comments, can it be like this ?

**Comment by tstrachota** [comment]
@dhlavac thanks, the tests look good now, but... Did you try to execute them prior to pushing the changes?

**Comment by jyejare** [comment]
QE ACK

Tests Performed:

```
1. List all the auth sources using hammer auth-source list
Assert: All the auth sources are listed

2. List auth sources with per-page option
Assert: Auth-sources listed correctly with per-page option

3. List auth sources with page option
Assert: Auth-sources listed correctly with page option

4. List auth sources of specific organization
Assert: Auth-sources listed correctly of a specific organization

5. List auth sources of specific location
Assert: Auth-sources listed correctly of a specific location

6. List auth sources with the search option
Assert: Auth-sources listed correctly with the search option
```

**Comment by dhlavac** [comment]
@tstrachota yes It works for me .. but yes i see something went wrong .. I will look at that

**Comment by dhlavac** [comment]
@tstrachota need adpidoc 1.18 for tests .. I have added it in new comit in PR

**Comment by tstrachota** [comment]
[test]

**Comment by tstrachota** [comment]
Great, thanks @dhlavac I suspect that some other tests, not related to your PR, might fail with the new apidoc export. But we'll see, let's wait for the results...

**Comment by mbacovsky** [comment]
I'll move the updated APIdoc to separate PR and fix the tests there.

**Comment by mbacovsky** [comment]
I've added new PR with updated apidoc and fixes for the failing tests: https://github.com/theforeman/hammer-cli-foreman/pull/365

Once the apidoc update is merged. we can rebase and merge this PR.

**Comment by mbacovsky** [comment]
@dhlavac, could you rebase please. The fixed tests and your commit updating API docs were merged in master.

**Comment by dhlavac** [comment]
@mbacovsky rebased

**Comment by tstrachota** [comment]
[test]

**Comment by tstrachota** [comment]
Works like a charm now. Thank you @dhlavac !

**Pull request merged**